src/test/java/com/example/productapi/service/ProductServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package com.example.productapi.service;
+
+import com.example.productapi.model.Product;
+import com.example.productapi.repository.ProductRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.hibernate.validator.internal.util.Contracts.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class ProductServiceTest {
+    @Mock
+    private ProductRepository repository;
+
+    @InjectMocks
+    private ProductService service;
+
+    private Product product;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        product = new Product(1L, "Laptop", 1200.00, "A high-end laptop");
```

Comment:
try to add Builder in Product and use it here

try 2 options
1. manual code builder
2. @Builder from Lombok and usage here